File: solutions-blind/python/005-rle/solution_test.py
from solution import encode, decode

assert encode("") == ""
assert encode("a") == "a1"
assert encode("aaabbc") == "a3b2c1"
assert encode("WWWWWWWWWWWWBWWWWWWWWWWWWBBB") == "W12B1W12B3"
assert decode("a3b2c1") == "aaabbc"
assert decode("W12B1W12B3") == "WWWWWWWWWWWWBWWWWWWWWWWWWBBB"
for bad in ("a", "a0"):
    try:
        decode(bad)
        assert False, bad
    except ValueError:
        pass
for s in ("aaabbc", "Hello", ""):
    assert decode(encode(s)) == s
print("ok")

File: solutions-blind/javascript/005-rle/solution.test.mjs
import test from "node:test";
import assert from "node:assert/strict";
import { encode, decode } from "./solution.mjs";

test("encode", () => {
  assert.equal(encode(""), "");
  assert.equal(encode("a"), "a1");
  assert.equal(encode("aaabbc"), "a3b2c1");
  assert.equal(encode("WWWWWWWWWWWWBWWWWWWWWWWWWBBB"), "W12B1W12B3");
});

test("decode", () => {
  assert.equal(decode("a3b2c1"), "aaabbc");
  assert.equal(decode("W12B1W12B3"), "WWWWWWWWWWWWBWWWWWWWWWWWWBBB");
  assert.throws(() => decode("a"));
  assert.throws(() => decode("a0"));
});

test("round-trip", () => {
  for (const s of ["aaabbc", "Hello", ""]) {
    assert.equal(decode(encode(s)), s);
  }
});
